src/pages/cart.jsx

```diff
@@ -23,13 +23,97 @@ import {
+    if(window.onwardApp) {
+      window.onwardApp.cartChanged(parseCheckout(checkout));
+    }
+  }, [checkout]);
+
+  React.useEffect(() => {
+    window.onwardStorefrontClient = {
+      fetchInsuranceProduct: async function(){
+        return await client.product.fetchByHandle('onward-package-protection').then((product) => {
+          return {
+            variants: product.variants.map(variant =>
+              ({
+                id: variant.id,
+                price: { amount: Number(variant.price.amount), currencyCode: variant.price.currencyCode }
+              })
+            )
+          };
+        });
+      },
+      fetchCart: async function(){
+        return await client.checkout.fetch(checkout.id).then(parseCheckout);
+      },
+      clearCart: async function() {
+        // clearCart(): Promise<Cart>;
+        return await client.checkout.removeLineItems(checkout.id, checkout.lineItems.map(item => item.id)).then(parseCheckout);
+      },
+      updateCartLines: async function(changes) {
+        // updateCartQuantities(changes: Record<string, number>): Promise<Cart>;
+        const response = await client.checkout.updateLineItems(
+          checkout.id,
+          changes.map(({ id, quantity }) => {
+            return {
+              id,
+              quantity,
+            };
+          }),
+        );
+
+        setCheckout(response);
+
+        return parseCheckout(response);
+      },
+      addCartLines: async function(changes) {
+        // applyCartLineChanges(changes: CartLineChange[]): Promise<Cart>;
+        return parseCheckout(await client.checkout.addLineItems(
+          checkout.id,
+          changes.map(({ variantId, quantity }) => {
+            return {
+              variantId,
+              quantity,
+            };
+          }),
+        ));
+      },
+    };
+
+    window.OnwardSettingsJson = {"selectedByDefault":true,"shopEnabled":true,"insuranceFeePercentage":"2.0","carbonOffsetPriceCents":10,"carbonOffsetsEnabled":true,"shopName":"onward-checkout-store","minimumInsurancePriceCents":null,"minimumInsurancePriceVariantId":null,"excludePriceInSubtotalDisplay":true,"smallerSwitch":true,"oneLine":false,"poweredByEnabled":true,"disclaimerEnabled":true,"showSubscriptionWording":true,"moneyFormat":"{{amount}}","checkoutButtonSelector":null,"themePrimaryColor":null,"ctaTextColor":null,"copyLine1":"{{reward_amount}} reward","copyLine2":"Get {{reward_amount}} store credit protect your package from theft, loss or damage.","disclaimerText":"*By deselecting shipping protection, onward-checkout-store is not liable for lost, damaged, or stolen items","rewardsEnabled":true,"rewardsPercentage":"5.0","nativeSubscriptionsEnabled":false,"brandLogoUrl":"https://john1.useonward.dev/public/brandings/127/brand_logo","brandLogoAltUrl":"https://john1.useonward.dev/public/brandings/127/brand_logo_alt","cartItemCountSelector":".cart-count-bubble span","cartSubtotalSelector":".totals__subtotal-value"};
```

Comment:
next step is pull in the settings dynamically

Reply:
Can define these in an async way just so it's closer to what a real-world example would be like? Would a headless shop serialize this into the page from the server, or (more likely) it'd be fetched from the server, right?

Reply:
My plan is to have the client fetch the headless js with a shop identifier and in that response we would inject the settings. So it would all be packages together in one response